test/leafnode_test.go

```diff
@@ -3228,3 +3228,82 @@ func TestClusterTLSMixedIPAndDNS(t *testing.T) {
 	// Make sure this works.
 	checkLeafNodeConnected(t, srvA)
 }
+
+// This will test for a bug in stream export/import with leafnodes.
+// https://github.com/nats-io/nats-server/issues/1332
+func TestStreamExportWithMultipleAccounts(t *testing.T) {
+	confA := createConfFile(t, []byte(`
+		listen: 127.0.0.1:-1
+		leafnodes {
+			listen: "127.0.0.1:-1"
+		}
+	`))
+	srvA, optsA := RunServerWithConfig(confA)
+	defer srvA.Shutdown()
+
+	bConfigTemplate := `
+		listen: 127.0.0.1:-1
+		leafnodes {
+			listen: "127.0.0.1:-1"
+			remotes = [
+			{
+				url:"nats://127.0.0.1:%d"
+				account:"EXTERNAL"
+			}
+			]
+		}
+		accounts: {
+		    INTERNAL: {
+		        users: [
+		            {user: good, password: pwd}
+		        ]
+			    imports: [
+		            {
+		                stream: {
+		                    account: EXTERNAL
+		                    subject: "foo"
+		                }
+		            }
+		        ]
+		    },
+		    EXTERNAL: {
+		        users: [
+		            {user: bad, password: pwd}
+		        ]
+		        exports: [{stream: "foo"}]
```

Comment:
To make it "fun" maybe you could have made the export "bar" and then publish on "bar" and make sure it is received (on "foo"). This will make sure that the LS+ is sent on "bar" and not ">" (or "foo"), no?

Reply:
Good idea, will do.

Reply:
You mean remap the foo on import with a prefix?

Reply:
Can be on prefix + "foo" for streams. But will add that to make sure we do prefix subj.

Reply:
Sorry, I thought that we were mapping.. for stream there is only prefix..

Reply:
For completeness, test a service too?